kafkakewl-domain/src/main/scala/com/mwam/kafkakewl/domain/validation/DeploymentsValidation.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.mwam.kafkakewl.domain.validation
+
+import com.mwam.kafkakewl.domain.*
+import com.mwam.kafkakewl.utils.CollectionExtensions.*
+
+object DeploymentsValidation {
+  import TopologyValidation.*
+
+  def validate(topologiesBefore: Topologies, deployments: Deployments): ValidationFailures = {
+    // First validate the deployments itself on its own
+    val deploymentValidation = validateDeployments(deployments)
+
+    // Then pretend we applied the new topologies and the removed ones and...
+    val topologiesAfter = topologiesBefore ++ deployments.deploy.toMapById -- deployments.delete
```

Comment:
minor nitpick `deployments.deploy` is a bit confusing to read, I had to check `Deployments` class to confirm that this is just Seq[Topology]

Reply:
agree, my goal was to make json/yaml more readable/writable. Do you have any other name ideas that look good both in the json/yaml and code? Alternatively we can use different names in json/yaml, but that sounds a bit overkill.